mistletoe/block_token.py

```diff
@@ -324,14 +324,14 @@ def read(cls, lines):
                 and not Quote.start(next_line)):
 
             # check if next_line starts List
-            list_pair = ListItem.parse_marker(next_line)
+            marker_tuple = ListItem.parse_marker(next_line)
             if (len(next_line) - len(next_line.lstrip()) < 4
-                    and list_pair is not None):
-                prepend, leader = list_pair
-                # non-empty list item
-                if next_line[:prepend].endswith(' '):
-                    # unordered list, or ordered list starting from 1
-                    if not leader[:-1].isdigit() or leader[:-1] == '1':
+                    and marker_tuple is not None):
+                _, leader, content = marker_tuple
+                # to break a paragraph, the first line may not be empty,
+                # and the list must be unordered or start from 1.
+                if not content.strip() == '':
+                    if not leader[0].isdigit() or leader in ['1.', '1)']:
```

Comment:
It may be, but I think this is more readable 😄 plus it won't accept just any number ending with a one.

Reply:
It may be, from a point of view. :) But the original condition mentioned won't accept e.g. `21` (plus whatever from `.` and `)` follows) either, or do I miss something here? :)

Reply:
No you're right, it just strips off the last character and compares the rest to "1", so that should work fine. I think spelling out the options is more readable though.

Reply:
OK :)